namui/src/namui/common/text/multiline_caret.rs

```diff
@@ -0,0 +1,186 @@
+use crate::{
+    namui::skia::GlyphIds,
+    system::text_input::{ArrowUpDown, KeyInInterest},
+    text::*,
+    *,
+};
+
+#[derive(Debug, Clone, Copy)]
```

Comment:
I think this struct and `LineTexts` should be refactored. I will do it later, not now.